### DISCUSSION TITLE
Objective function choice for FBA for non-malignant tissue

### DISCUSSION THREAD

**Discussion by manas-kohli** [SysBioChalmers/Human-GEM#833] [Q&A] [answered]
Hello Human1 team!

I want to compare metabolic fluxes between non-malignant and tumour tissue and so basically have run FBA so far for many TCGA samples. I'd like to now run this for GTEx samples for the appropriate tissue type to serve as a control so to speak. For this I think it would be better to choose a different objective function and one commonly cited in the literature is ATP synthesis. Along those lines I had two questions:
 
1. What would be a good reaction/reactions to choose for ATP synthesis for non-malignant tissues? For the cancer case optimising for biomass is a justified and straightforward choice but I don't see a simple reaction/set of reactions for ATP synthesis. I also further looked at some of my models and noticed that the reaction MAR03904 i.e. ATP hydrolysis is missing which is what I also thought might make a good choice. 

2. I had questions regarding this reaction as well as it's not reversible and the opposite reaction is what I'd aim to probably use as an objective function. Regardless I found it a bit strange the ATP hydrolysis reaction was ommitted from my models. I confirmed the initial prepData model for ftINIT had this reaction. Also when I looked at the complete metabolic tasklist for my models, they passed the initial check of being able to perform Aerobic rephosphorylation of ATP from glucose and also the same for fatty acids. Perhaps this reaction not being present in my models is normal? If it isn't I'm attaching a TPM file from a small subset GTEx that I used. 
[GTEx_subset_TPM.csv](https://github.com/user-attachments/files/16051981/GTEx_subset_TPM.csv) 

Thanks and Best Regards,
Manas

**Comment by JonathanRob** [comment]
Hi @manas-kohli!

One thing to keep in mind is that comparing fluxes between models that are using different objective functions is likely to primarily identify differences associated with a change in objective function, rather than on model structural differences associated with malignancy. This is not to say that it's incorrect, only to be aware of how this may affect the results.

1. The ATP hydrolysis reaction [MAR03964](https://metabolicatlas.org/explore/Human-GEM/gem-browser/reaction/MAR03964) (I believe you had a typo in your rxn ID above) could be an option to use as the objective. Depending on how your models were generated, this reaction is likely excluded because it has no gene associated with it and/or it doesn't provide any energetic/resource benefit to the network since it just hydrolyzes ATP without coupling it with proton pumping or some other reaction. You can always temporarily add it back to the model for the purposes of FBA, since the model should at least have the components required to carry flux through the reaction once added.

2. Along the lines of what I described above, this ATP hydrolysis reaction is not really necessary for anything - it is essentially an energy dump that wastes chemical potential. If you reverse the reaction, then you suddenly give the cell a free energy source (phosphorylating ADP with no energy input), which would be highly unrealistic and not provide any useful results. The reason for setting the ATP hydrolysis reaction (in its normal forward direction) as the objective function is that in order to maximize flux through the reaction, the cell needs to maximize its "production" (phosphorylation) of ATP to provide the input substrates for the reaction.

**Comment by IVANDOMENZAIN** [comment]
@manas-kohli think about using random sampling approaches (given suitable constraints for your case), rather than focusing on individual FBA solutions. Especially if you are evaluating models generated from bulk-gene expression samples, that all together represent lumped/average expression in tissues/tumours. Then you will get an exploration of the "metabolic landscape" of the cells in your tissue, from which you might get relevant statistical information and also fine-grained reaction level info for the identified conditions of interest in your flux sampling.

**Comment by manas-kohli** [comment]
Thank you both for your suggestions/answers. These are very useful and I really appreciate the prompt responses :)

**Comment by manas-kohli** [comment]
Hey there, 

I wanted to follow up a bit on this as I've been having some issues/concerns.

1. For the random sampling - I performed random sampling for a few ecModels I created. The only thing I modified was to keep the protein pool lower bound constant and not set to -inf for every sampling. From this, I obtained a bunch of solutions but for many of them, only a handful of reactions actually have flux. I assume this is somewhat normal as for a random objective function, likely not a lot of flux is needed to route to these reactions but I still wanted to clarify if this is the case or if I'm doing something wrong here. Is there also a way for using randomsampling to fix the objective function and explore different possible flux solutions. So for example, still setting the biomass to be the objective function and coming up with different solutions. I assume this isn't the case because FBA is just a linear programming problem but still thought I may ask. Along those lines I'm a bit confused as to what's the meaningful information I can truly get from random sampling and whether solutions that just contain about 20-30 reactions with non-zero flux can be discarded. 

2. Comparing non-malignant and malignant tissue - from the literature it seems evident that people repeatedly cite using the ATP hydrolysis reaction as a suitable objective function. I tried this for the cytoplasmic ATP hydrolysis reaction which I added (mentioned in the above discussion) and also the mitochondrial one (MAR06916 i.e. oxphos).  However, when I do this for my models, the FBA solutions it obtains seem to be a bit trivial. Somewhat predictably, the models just import glucose, feed it into glycolysis, the TCA cycle and then oxphos to satisfy the objective function. This seems fine but obviously cells are not only doing this. For example when looking at the flux solutions for this, there's no real flux through any of the lipid pathways. I'm a bit confused I suppose on the interpretation on this if it makes sense. I can see differences in the GEM construction between normal and cancer tissue but I suppose it's a bit difficult to say what the metabolic flux differences are. The biomass objective function is nice in that it incorporates many varied metabolic processes and so gives u a snapshot of metabolism as a whole. However, for the so called 'maintenance phenotype', I'm finding the ATP hydrolysis reaction set as the objective function a bit lacking. Do you have any possible advice as how it would be best for me to compare cancer vs normal tissue from a flux perspective? Is there a sort of aritificial 'maintenance' reaction in the GEM that I can use as a surrogate. While cells aren't looking to optimise their biomass, I suppose they still need to route flux into this pathway for general repair e.g. making proteins, lipids and stuff like that. However, if I were to set bounds on the biomass reaction, I'm not sure what would be suitable bounds to use. Anything in the ballpark can also be fine where I fix the bounds for biomass and then optimise ATP hydrolysis (don't know if any of this makes actual sense or if I'm on a tangent somewhere). Otherwise, would it be best for me to probably cut my losses and look at comparing cancer models between themselves rather than the whole metabolic reprogramming aspect of it.

I appreciate these are quite loaded questions, any advice particularly on the randomsampling stuff would be very much appreciated. Thanks for your time :)

**Reply by edkerk** [reply, marked as answer]
1. Random sampling results can differ drastically dependent on the constraints that are set. As you also hint towards, it often makes sense to constrain the model for instance with a particular biomass production (set the lower and upper bounds to the same value or with a bit of flexibility). What random sampling can tell you is: given the constraints that are applied to the model, what is the likeliness of flux going through each reaction? You could for instance (a) set nutrient uptake rates to values that are experimentally measured, or reasonable estimates; (b) maximum growth as objective function, and now fix this value (lower bound = upper bound), and also fix the nutrient rates; (c) maximize the ATP turnover (flux to maintenance reaction) and fix this value; (d) in case of ecModels, minimize the total protein usage and fix this value; (e) perform random sampling. Now all of the randomly sampled flux distributions will have the same nutrient uptake, growth rate, ATP turnover and protein usage, but there will still be multiple solutions possible, from which you can determine the means and standard deviations for each reaction. If you do the same for a different condition/tissue/patient, and compare the means + standard deviations between them (potentially after normalizing all fluxes to the same growth rate, glucose uptake, or something else), you can apply t-tests for each reaction to compare the means and standard deviations.

If you only have a few reactions active in most of the simulations, you should probably constrain your model more, so that each random sample is a closer representation of reality, and not only a theoretical random solution where the model is left unconstrained.

2. I have limited experience in simulating human GEMs, but indeed only maximizing ATP turnover would result in nothing more than just generating ATP. You might want to consider to fix a non-zero flux through the biomass reaction, to reflect that even stationary cells require turnover of macromolecules. Then, it would be best to split the biomass reaction, as there is likely far less (if any) DNA turnover, in comparison to RNA and protein turnover. But you also seem to have thought along those lines, so I'm afraid I don't have much more to add there.
